OpenAPI/kiota/models.md

```diff
@@ -21,6 +21,27 @@ All models declared inline with an operation will be generated under the namespa
 
 Models in an [allOf](https://spec.openapis.org/oas/latest.html#composition-and-inheritance-polymorphism) schema declaration will inherit from each other. Where the uppermost type in the collection is the greatest ancestor of the chain.
 
+### allOf interpretation rules
+
+The following table describes how kiota will project any schema with allOf entries when processing an API description:
+
+| Number of Properties | Number of allOf entries without a reference / inline | Number of allOf entries with a reference | Total number of allOf entries | Result |
+| -------------------- | ---------------------------------------- | ------------------------------------------- | ----------------------------- | ------ |
+| 0 | 0 | 0 | 0 | Ignored/Invalid |
+| 0 | 0 or 1 | 0 or 1 | 1 | Ignored, will process the only allOf entry instead and use the original schema's description and name. |
+| 1 or more | 0 | 0 | 0 | Class/interface without a parent type. |
+| 0 | 1 | 1 | 2 | Class/interface with properties from the inline schema and a parent type from the referenced schema. |
+| 1 or more | 0 | 1 | 1 | Class/interface with properties from the current schema and a parent type from the referenced schema. |
+| 1 or more | 1 | 0 | 1 | Class/interface with properties from the current schema and a parent type from the inline schema. |
```

Comment:
How do you name the parent type?

Reply:
based of the root schema, depending on whether it's a ref or inline itself (the logic here is outside of this block)